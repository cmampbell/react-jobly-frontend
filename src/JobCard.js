import React from 'react'
import { useOutletContext } from 'react-router-dom';
import { Card, CardBody, CardTitle, CardText, Button} from 'reactstrap';

const JobCard = ({job}) => {
    // applications is a set of job.id that user has applied to, saved in state on <App />
    const { applyToJob, applications } = useOutletContext()
    return (
        <Card className="text-start" color="info" outline style={{ width: '80vw' }}>
            <CardBody>
                <CardTitle tag="h5">
                    {job.title} at {job.companyName}
                </CardTitle>
                <CardText>
                    Salary: {job.salary}
                </CardText>
                <CardText>
                    Equity: {job.equity ? job.equity : 'None'}
                </CardText>
                {/* if the job id is in application, show application submitted button */}
                {!applications.has(job.id)  ? 
                <Button onClick={() => applyToJob(job.id)}> Submit Application! </Button>
                : <Button>Application Submitted!</Button>
            }
            </CardBody>
        </Card>
    )
}

export default JobCard;