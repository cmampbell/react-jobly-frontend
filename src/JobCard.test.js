import { render } from '@testing-library/react';
import JobCard from './JobCard';

// mock react-router-dom named exports for this component
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useOutletContext: function () {
    return {
      applyToJob: function () {
        return null
      },
      applications: new Set()
    }
  }
})
);

describe('Smoke and snapshot CompanyCard test', () => {
  it('should render without crashing', () => {
    render(<JobCard job={{
      title: 'test',
      companyName: 'test',
      salary: '50000',
      equity: '.03'
    }} />
    );
  });
  it('should match snapshot', async () => {
    const { asFragment, findByText } = render(
    <JobCard job={{
      title: 'test',
      companyName: 'test',
      salary: '50000',
      equity: '.03'
    }}/>);
    expect(asFragment()).toMatchSnapshot();
    expect(await findByText('Submit Application!')).toBeInTheDocument();
  })
})
