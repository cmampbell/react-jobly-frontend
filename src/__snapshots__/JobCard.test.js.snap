// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot CompanyCard test should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-start card border-info"
    style="width: 80vw;"
  >
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        test at test
      </h5>
      <p
        class="card-text"
      >
        Salary: 50000
      </p>
      <p
        class="card-text"
      >
        Equity: .03
      </p>
      <button
        class="btn btn-secondary"
        type="button"
      >
         Submit Application! 
      </button>
    </div>
  </div>
</DocumentFragment>
`;
